pkg/apis/kubermatic/v1/external_cluster.go

```diff
@@ -64,6 +64,131 @@ type ExternalClusterSpec struct {
+	ManifestReference *providerconfig.GlobalSecretKeySelector `json:"manifestReference,omitempty"`
+	CloudSpec         *KubeOneCloudSpec                       `json:"cloudSpec,omitempty"`
+}
+
+type KubeOneCloudSpec struct {
+	AWS          *KubeOneAWSCloudSpec          `json:"aws,omitempty"`
+	GCP          *KubeOneGCPCloudSpec          `json:"gcp,omitempty"`
+	Azure        *KubeOneAzureCloudSpec        `json:"azure,omitempty"`
+	Digitalocean *KubeOneDigitaloceanCloudSpec `json:"digitalocean,omitempty"`
+	Openstack    *KubeOneOpenstackCloudSpec    `json:"openstack,omitempty"`
+	Packet       *KubeOnePacketCloudSpec       `json:"packet,omitempty"`
+	Hetzner      *KubeOneHetznerCloudSpec      `json:"hetzner,omitempty"`
+	VSphere      *KubeOneVSphereCloudSpec      `json:"vsphere,omitempty"`
+	Nutanix      *KubeOneNutanixCloudSpec      `json:"nutanix,omitempty"`
+}
+
+// KubeOneAWSCloudSpec specifies access data to Amazon Web Services.
+type KubeOneAWSCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
+
+	AccessKeyID     string `json:"accessKeyID,omitempty"`
+	SecretAccessKey string `json:"secretAccessKey,omitempty"`
+}
+
+// KubeOneGCPCloudSpec specifies access data to GCP.
+type KubeOneGCPCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
+
+	ServiceAccount string `json:"serviceAccount,omitempty"`
+}
+
+// KubeOneAzureCloudSpec specifies access credentials to Azure cloud.
+type KubeOneAzureCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
+
+	TenantID       string `json:"tenantID,omitempty"`
+	SubscriptionID string `json:"subscriptionID,omitempty"`
+	ClientID       string `json:"clientID,omitempty"`
+	ClientSecret   string `json:"clientSecret,omitempty"`
+}
+
+// KubeOneDigitaloceanCloudSpec specifies access data to DigitalOcean.
+type KubeOneDigitaloceanCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
+
+	Token string `json:"token,omitempty"` // Token is used to authenticate with the DigitalOcean API.
+}
+
+// KubeOneOpenstackCloudSpec specifies access data to an OpenStack cloud.
+type KubeOneOpenstackCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
+
+	AuthURL  string `json:"authURL,omitempty"`
+	Username string `json:"username,omitempty"`
+	Password string `json:"password,omitempty"`
+
+	// project, formally known as tenant.
+	Project string `json:"project,omitempty"`
+	// project id, formally known as tenantID.
+	ProjectID string `json:"projectID,omitempty"`
+
+	Domain string `json:"domain,omitempty"`
+}
+
+// KubeOneVSphereCloudSpec credentials represents a credential for accessing vSphere.
+type KubeOneVSphereCloudSpec struct {
```

Comment:
There's also `allowInsecure` option for vSphere clusters.

Reply:
okay, what is the KubeOne env var for this?

Reply:
My bad, this is not an option in KubeOne.